src/core/manager.c

```diff
@@ -2749,6 +2749,10 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                 r = unit_deserialize(u, f, fds);
                 if (r < 0)
                         goto finish;
+
+                r = unit_realize_cgroup(u);
```

Comment:
Hmm good question, actually I would answer that resource controls are unit properties that can be tuned dynamically without the need to restart the unit.

OTOH, the Exec stuff is related to how a service is started hence the `restart` command seems more logical in this case.